Add a Bootstrap Navbar

bobsilverberg opened the pull request:
Fixes #76 

A couple of notes:

1. As I mentioned in today's standup, I think we should remove `profile` from the component state of `App` and move it to the Redux store. As that will require some extra work, including the creation of a reducer, I think it makes sense to defer that to a follow up.

2. I wasn't sure what to do about the "logging out" message, so I just left it in the `App` component for now. I'm not sure if we need that at all, or if it was just there to demonstrate that it works.

Logged Out:

![screenshot 2019-01-30 15 40 14](https://user-images.githubusercontent.com/142755/52011221-5ffb0180-24a5-11e9-8982-3564045ca2b5.png)

Logged In:

![screenshot 2019-01-30 15 41 15](https://user-images.githubusercontent.com/142755/52011272-81f48400-24a5-11e9-872b-e9c7b28dd014.png)

bobsilverberg commented:
One more note, there's a TS error on the test in `src/components/App/index.spec.tsx` because I'm using `root.instance().logOut` and TS doesn't recognize that that exists. I'm not sure how to fix this - advice appreciated.

kumar303 commented on src/components/Navbar/styles.module.scss:
```diff
@@ -0,0 +1,3 @@
+.Navbar {
+  justify-content: space-between;
```
Since `justify-content` only works for flex, how about explicitly setting `display: flex`? Otherwise, it seems a bit risky to rely on an implicit flex display.

bobsilverberg replied:
Because this is a `react-bootstrap` component, we know it will have `display: flex` on it. That's the way these bootstrap classes work, which is why I left it off. But I see no reason not to explicitly set it, so I will.

willdurand commented on src/components/Navbar/index.spec.tsx:
```diff
@@ -0,0 +1,58 @@
+      const onLogOut = () => null;
+      const root = render({ onLogOut, profile: createProfile() });
+
+      expect(root.find('.Navbar-logOut')).toHaveProp('onClick', onLogOut);
```
Isn't it `jest.fn()`?

willdurand commented on src/components/Navbar/index.tsx:
```diff
@@ -0,0 +1,37 @@
+          ) : null}
+          {profile ? (
+            <Button className="Navbar-logOut" onClick={onLogOut}>
+              Log out
```
Can we defer the i18n stubs in a new issue? I thought we wanted to give a more react-y API a try (Lingui?), so this could end up not being a `gettext` function.

bobsilverberg replied:
I'm fine with either. I suppose it's not going to be too big a deal to find and replace all instances of `gettext` with something else if/when we decide to go in a different direction, but I do agree that switching from a function to a tag (Lingui style), would be a bit more work.

I think we believe that we probably will not internationalize this app at all, in which case whatever is the least amount of work at this point is probably what we should do, and I think that's probably using a function.

I'll wait to hear back from both of you before proceeding with this.

bobsilverberg commented on src/components/Navbar/index.tsx:
```diff
@@ -0,0 +1,37 @@
+        <Navbar.Brand href="#">addons-code-manager</Navbar.Brand>
+        <Nav>
+          {profile ? (
+            <Nav.Link className="Navbar-username">{profile.name}</Nav.Link>
```
I made it a `Nav.Link` because that's what works out of the box with Bootstrap to make it appear the way we want. But it doesn't need to be a link so I've changed it to use `Navbar.Text` and some custom styling.

bobsilverberg commented on src/components/App/index.spec.tsx:
```diff
@@ -29,7 +31,8 @@ describe(__filename, () => {
     const root = render();
 
-    expect(root).toHaveClassName(styles.container);
+    expect(root.find(Navbar)).toHaveProp('onLogOut', root.instance().logOut);
```
That makes sense. I'll remove the assertion for now.

bobsilverberg commented:
Thanks for the review @kumar303. I've addressed all of the issues except for the one about i18n. I'm awaiting the result of the discussion at https://github.com/mozilla/addons-code-manager/pull/87#discussion_r252685088 for that.

willdurand commented on src/components/Navbar/index.tsx:
```diff
@@ -0,0 +1,37 @@
+          ) : null}
+          {profile ? (
+            <Button className={styles.logOut} onClick={onLogOut}>
+              Log out
```
On this basis, then 👍
